python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+                      unit="pixel"):
+        """Return an image centered on ra and dec (in degrees) with dimensions
+        height and width (in arcseconds by default).
+
+        - Determine approximate pixels per arcsec in the image by
+             calculating the length of line from the upper right corner of
+             the image to the lower left corner in pixels and arcsecs.
+             (This will fail at or very near the pole.)
+        - Use that to define a box for the cutout.
+        - Trim the box so it is entirely within the source image.
+
+        Parameters
+        ----------
+        src_img : `afwImage.Exposure`
+            the source image.
+        data_id : `dict`
+            the image id.
+        ra : float
+        dec :  float
+            in degrees.
+        height, width : float, float
+            Height and width are in pixels, by default.
+
+        Returns
+        -------
+        cutout: `afwImage.Exposure`
+
+        """
+        self._log.debug("apply_cutout %f %f %f %f", ra, dec, width, height)
+        # False: do not remove FITS keywords from metadata
+        wcs = None
+        if isinstance(src_img, afwImage.Exposure):
+            wcs = src_img.getWcs()
+        elif data_id:
+            # Get the metadata for the source image.
+            metadata = self._metadata_from_data_id(data_id)
+            wcs = afwGeom.makeSkyWcs(metadata, strip=False)
+        if wcs is None:
+            raise Exception("wcs missing in source image")
+        radec = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
+        xy_wcs = wcs.skyToPixel(radec)
+        xy_center_x = xy_wcs.getX()
+        xy_center_y = xy_wcs.getY()
+        self._log.debug("ra=%f dec=%f xy_center=(%f,%f)",
+                        ra, dec, xy_center_x, xy_center_y)
+        if unit == "arcsec":
+            # pixel scale is defined as Angle/pixel
+            ps = wcs.getPixelScale().asArcseconds()
+            if ps != 0:
+                width = width / ps
+                height = height / ps
+            else:
+                self._log.debug("pixel scale = 0!")
+        cutout = self.cutout_from_src(src_img, xy_center_x, xy_center_y, width,
+                                 height, wcs)
+
+        return cutout
+
+    def _metadata_from_data_id(self, data_id):
+        # Return the metadata for the query results in qResults.
+        metadata = self._butler.get(self._imagedataset_md(), dataId=data_id)
+        return metadata
+
+    def _image_from_butler(self, data_id, bbox=None):
+        # Retrieve the image through the Butler using data id.
+        self._log.debug("_image_from_butler data_id:{}".format(data_id))
+        if bbox:
+            image = self._butler.get(self._imagedataset_sub(), bbox=bbox,
+                                     dataId=data_id, immediate=True)
+        else:
+            image = self._butler.get(self._imagedataset_type, dataId=data_id)
+        return image
+
+    def _imagedataset_md(self):
```

Comment:
Both of these functions only seem to be called in one place, and they are just one line.  Maybe just inline these into the callers (that way you don't need to find the function to know what it is / does).